backend/capellacollab/projects/toolmodels/modelsources/git/github/interface.py

```diff
@@ -0,0 +1,197 @@
+            },
+            timeout=2,
+        )
+    response.raise_for_status()
+    return response.json()["workflow_runs"]
+
+
+def get_artifact_from_job_as_content(
+    project_id: str,
+    job_id: str,
+    trusted_path_to_artifact: str,
+    git_model: git_models.DatabaseGitModel,
+    git_instance: settings_git_models.DatabaseGitInstance,
+) -> bytes:
+    return get_artifact_from_job(
+        project_id, job_id, trusted_path_to_artifact, git_model, git_instance
+    ).encode()
+
+
+def get_artifact_from_job_as_json(
+    project_id: str,
+    job_id: str,
+    trusted_path_to_artifact: str,
+    git_model: git_models.DatabaseGitModel,
+    git_instance: settings_git_models.DatabaseGitInstance,
+) -> dict:
+    return json.loads(
+        get_artifact_from_job(
+            project_id,
+            job_id,
+            trusted_path_to_artifact,
+            git_model,
+            git_instance,
+        )
+    )
+
+
+def get_artifact_from_job(
+    project_id: str,
+    job_id: str,
+    trusted_path_to_artifact: str,
+    git_model: git_models.DatabaseGitModel,
+    git_instance: settings_git_models.DatabaseGitInstance,
+) -> str:
+    response = requests.get(
+        f"{git_instance.api_url}/repos{project_id}/actions/runs/{job_id}/artifacts",
+        headers={
+            "Authorization": f"token {git_model.password}",
+            "X-GitHub-Api-Version": "2022-11-28",
+            "Accept": "application/vnd.github+json",
+        },
+        timeout=2,
+    )
+    response.raise_for_status()
+    artifact_id = response.json()["artifacts"][0]["id"]
+
+    if response.json()["artifacts"][0]["expired"] == "true":
+        raise fastapi.HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail={
+                "err_code": "ARTIFACT_EXPIRED",
+                "reason": (
+                    "The latest artifact you are requesting expired. Please rerun your pipline and contact your administrator."
+                ),
+            },
+        )
+    artifact_response = requests.get(
+        f"{git_instance.api_url}/repos{project_id}/actions/artifacts/{artifact_id}/zip",
+        headers={
+            "Authorization": f"token {git_model.password}",
+            "X-GitHub-Api-Version": "2022-11-28",
+            "Accept": "application/vnd.github+json",
+        },
+        timeout=2,
+    )
+    artifact_response.raise_for_status()
+
+    return get_file_content(artifact_response, trusted_path_to_artifact)
+
+
+def get_file_content(
+    response: requests.Response, trusted_file_path: str
+) -> str:
+    with zipfile.ZipFile(io.BytesIO(response.content)) as zip_ref:
+        temp_dir = tempfile.mkdtemp()
+        zip_ref.extractall(temp_dir)
+        with open(
+            temp_dir + "/" + trusted_file_path.split("/")[-1],
+            encoding="utf-8",
+        ) as file:
+            content = file.read()
+        shutil.rmtree(temp_dir)
+        return content
```

Comment:
Writing to the disk is slow. Instead, you can do something like this: 
```py
with zipfile.ZipFile(io.BytesIO(response.content)), 'r') as zf:
        # Get the list of files in the zip
        file_list = zf.namelist()

        # Create a dictionary to store the unzipped data
        unzipped_data = {}

        # Go through each file in the zip
        for filename in file_list:
            # Read each file's data into memory
            with zf.open(filename, 'r') as file:
                unzipped_data[filename] = file.read()

        return unzipped_data
```

You can also modify the code snipped to just read the requested file.